### PR TITLE
Validate serial on player join

### LINKED ISSUE #3763: Server should reject empty serials
### Describe the bug

@ArranTuna reported this, but can't post it here (likely due to the world 'serial' being flagged).

If a user connects to a server while not having a serial (which is rare - but plausible) the server should reject the player. At the moment it seems like this is not happening.

Servers could of course do a check themselves in Lua, but it's still a good idea to incorporate a check like this.

### Steps to reproduce

N/A

### Version

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not security related.

### PR DESCRIPTION
**Pull request by Fernando-A-Rocha**
Fixes #3763 by preventing the player from connecting when serial is invalid: client is disconnected with error code `CD44 Serial verification failed`

[getPlayerSerial](https://wiki.multitheftauto.com/wiki/GetPlayerSerial) says that

> you should only trust the value returned by this function when used serverside

So this validation makes sense. With it, we can trust the value. The [Serial page](https://wiki.multitheftauto.com/wiki/Serial) mentions that it is always 32 characters long with letters and numbers. The letters are only A-F

Questions:

1. Why could [getPlayerSerial](https://wiki.multitheftauto.com/wiki/GetPlayerSerial) return false? In other words, why would a player not have a serial on a server?

2. Why does getPlayerSerial serverside accept a second argument (index) that, if 1 and not 0(which is default) returns the "extra serial" which is a string of max 64 chars? 

https://github.com/multitheftauto/mtasa-blue/blob/733683d70dc037fdcbb256fb17d86e93bfedd239/Server/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp#L707-L722

https://github.com/multitheftauto/mtasa-blue/blob/733683d70dc037fdcbb256fb17d86e93bfedd239/Server/mods/deathmatch/logic/net/CNetBuffer.cpp#L720

@Dutchman101

### CODE REVIEW

File: Server/mods/deathmatch/logic/CGame.cpp [hunk 1]
```diff
@@ -1781,6 +1782,18 @@ void CGame::Packet_PlayerJoinData(CPlayerJoinDataPacket& Packet)
         }
 #endif
 
+        // Prevent player from connecting if serial is invalid
+        std::regex serialRegex("^[A-F0-9]{32}$");
+        if (!std::regex_match(strSerial, serialRegex))
+        {
+            // Tell the console
+            CLogger::LogPrintf("CONNECT: %s failed to connect (Invalid serial) (%s)\n", szNick, pPlayer->GetSourceIP());
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CGame.cpp]: Does `GetSourceIP()` return `std::string`/`SString` or `const char*`? If the former then you need to use `.c_str()`

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Server/mods/deathmatch/logic/CGame.cpp (hunk 1 above)
Comment (Fernando-A-Rocha) [reply Server/mods/deathmatch/logic/CGame.cpp]: sorry that was incorrect, check now

**Review by TracerDS — APPROVED** [review]

**Comment by Dutchman101** [comment]
Thanks for this, even though servers using admin resource (revision [eb14427](https://github.com/multitheftauto/mtasa-resources/commit/eb144275b1cb875a4f088c0b3afa366d1162fc2e)) or higher are already protected, and AC backend performs checks too, having this in MTA Server (mtasa-blue) as well reduces the points of failure and overhead, making it more reliable and secure.

**Comment by Fernando-A-Rocha** [comment]
I think having this in mtasa-blue makes sense indeed. People may not use the admin resource in their servers

**Comment by CrosRoad95** [comment]
Wiki page https://wiki.multitheftauto.com/wiki/Serial should contain information about regex that matches serial with simple snippet example in lua. People may want to use it in external tools too so good to know it.

**Comment by Fernando-A-Rocha** [comment]
> Wiki page https://wiki.multitheftauto.com/wiki/Serial should contain information about regex that matches serial with simple snippet example in lua. People may want to use it in external tools too so good to know it.

Also, now the function will always return a string, never false. We can update wiki @Dutchman101

**Comment by Fernando-A-Rocha** [comment]
Regular users cannot edit the Serial page, and tbh let's edit the getPlayerSerial that'll be good.

**Pull request merged**